### PR TITLE
upgrade ingress nginx version

### LINKED ISSUE #4548: Gateway deployment failure due to K8s v1.22+ api removal
**Describe the Bug**
In Kubernetes v1.22, there are some API groups completely removed, https://kubernetes.io/blog/2021/07/14/upcoming-changes-in-kubernetes-1-22/#api-changes. After KubeSphere v3.2.0 release, we found there are still some compatibility issues left. In this article, https://kubesphere.com.cn/forum/d/6322-kubesphere, Gateway deployment still use `v1beta1/Ingress`, need to migrate to new API group `networking.k8s.io/v1`.

**Versions Used**
KubeSphere: v3.2.0
Kubernetes:  v1.22.1

**How To Reproduce**
1.  Enable gateway for a project.
2. Check the error log of the gateway

**Expected behavior**

/kind bug

### PR DESCRIPTION
**Pull request by chaunceyjiang**
### What type of PR is this?

/kind bug

### What this PR does / why we need it:
ingress nginx controller deployment failure due to K8s v1.22+ api removal

### Which issue(s) this PR fixes:

Fixes #4548 #4486 

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs
upgrade ingress nginx version
https://github.com/kubernetes/ingress-nginx/blob/helm-chart-4.0.13/Changelog.md
```

### CODE REVIEW

**Comment by RolandMa1986** [comment]
@chaunceyjiang It seems like the nginx-ingress 1.1.0 don't compatible with the old nginx 0.48.x. We can't simply change the downloaded helm chart version.

**Comment by RolandMa1986** [comment]
@chaunceyjiang Please check the [doc](https://github.com/kubernetes/ingress-nginx/blob/main/docs/index.md#i-have-only-one-instance-of-the-ingress-nginx-controller-in-my-cluster-what-should-i-do-).  in the v1.x release. we need add the following flags to let ingress controller monitoring all ingress.

>In this case, you need to make your controller aware of the objects. If you have any Ingress objects that don't yet have either the .spec.ingressClassName field set in their manifest, or the ingress annotation (kubernetes.io/ingress.class), then you should start your Ingress-NGINX controller with the flag --watch-ingress-without-class=true.

> You can configure your Helm chart installation's values file with .controller.watchIngressWithoutClass: true.

**Comment by chaunceyjiang** [comment]
Nginx Ingress - breaking change, Ingress.class now required
image

**Comment by chaunceyjiang** [comment]
> @chaunceyjiang Please check the [doc](https://github.com/kubernetes/ingress-nginx/blob/main/docs/index.md#i-have-only-one-instance-of-the-ingress-nginx-controller-in-my-cluster-what-should-i-do-). in the v1.x release. we need add the following flags to let ingress controller monitoring all ingress.
> 
> > In this case, you need to make your controller aware of the objects. If you have any Ingress objects that don't yet have either the .spec.ingressClassName field set in their manifest, or the ingress annotation (kubernetes.io/ingress.class), then you should start your Ingress-NGINX controller with the flag --watch-ingress-without-class=true.
> 
> > You can configure your Helm chart installation's values file with .controller.watchIngressWithoutClass: true.

ok

**Comment by chaunceyjiang** [comment]
> Nginx Ingress - breaking change, Ingress.class now required image

Turns out the fix was very simple, you just needed to ensure that your ingress annotations included the line kubernetes.io/ingress.class: "nginx" to specify the IngressClass. 

image
image

**Comment by RolandMa1986** [comment]
@chaunceyjiang if users are upgradding from an older version, all the ingresses needed to be updated manually. So it would be better to add ".controller.watchIngressWithoutClass: true" to let it works like the old way.

**Comment by chaunceyjiang** [comment]
> 

ok
you are right . I will add the parameter flag `--watch-ingress-without-class=true`

**Comment by chaunceyjiang** [comment]
> > @chaunceyjiang if users are upgradding from an older version, all the ingresses needed to be updated manually. So it would be better to add ".controller.watchIngressWithoutClass: true" to let it works like the old way.
> 
> ok you are right . I will add the parameter flag `--watch-ingress-without-class=true`

image

image

**Comment by chaunceyjiang** [comment]
so `kubesphere/nginx-ingress-controller:v1.1.0` is required

File: config/watches.yaml [hunk 1]
```diff
@@ -3,6 +3,8 @@
   version: v1alpha1
   kind: Nginx
   chart: /var/helm-charts/ingress-nginx
+  overrideValues:
```
Comment (RolandMa1986) [inline config/watches.yaml:6]: I would recommend changing the Nginx CR directly, the file located under:
https://github.com/kubesphere/kubesphere/blob/master/config/gateway/templates/nginx-ingress.yaml
The  OverrideValues are used when overriding values with environment variables.

**Comment by chaunceyjiang** [comment]
image
image

**Comment by RolandMa1986** [comment]
/lgtm
@chaunceyjiang btw, The docker image version will be overridden by the ks-installer:
https://github.com/kubesphere/ks-installer/blob/master/roles/download/defaults/main.yml

**Comment by chaunceyjiang** [comment]
> /lgtm @chaunceyjiang btw, The docker image version will be overridden by the ks-installer: https://github.com/kubesphere/ks-installer/blob/master/roles/download/defaults/main.yml

ok

https://github.com/kubesphere/ks-installer/pull/1872

**Pull request merged**